hibernate-core/src/test/java/org/hibernate/orm/test/id/uuid/SybaseASEUUIDTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.id.uuid;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import org.hibernate.dialect.SybaseASEDialect;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.RequiresDialect;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.util.uuid.SafeRandomUUIDGenerator;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.nio.ByteBuffer;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Jan Schatteman
+ */
+@RequiresDialect(value = SybaseASEDialect.class)
+@DomainModel(annotatedClasses = { SybaseASEUUIDTest.Book.class })
+@SessionFactory
+public class SybaseASEUUIDTest {
+
+	private UUID uuid;
+
+	@BeforeEach
+	void setUp(SessionFactoryScope scope) {
+		this.uuid = scope.fromTransaction( session -> {
+			UUID uuid = UUID.randomUUID();
+			// Create a UUID with trailing zeros
+			while ( SafeRandomUUIDGenerator.isSafeUUID(uuid) ) {
+				uuid = UUID.randomUUID();
+			}
+			final Book book = new Book(uuid, "John Doe");
```

Comment:
Just use a static UUID.

```suggestion
			final Book book = new Book( UUID.fromString("53886a8a-7082-4879-b430-25cb94415b00") , "John Doe");
```